File: attached_assets/Starter_code/Dotnet_starter_code/Insurance.Api/Controllers/UnderwritingController.cs
using Insurance.Api.Services;
using Microsoft.AspNetCore.Mvc;

namespace Insurance.Api.Controllers;

[ApiController]
[Route("api/underwriting")]
public class UnderwritingController : ControllerBase
{
    private readonly UnderwritingService _service;
    public UnderwritingController(UnderwritingService service) { _service = service; }

    public record EvaluateRequest(int Age, string Product, int PriorClaims);

    [HttpPost("evaluate")]
    public ActionResult<object> Evaluate([FromBody] EvaluateRequest body)
    {
        var result = _service.Evaluate(body.Age, body.Product, body.PriorClaims);
        return Ok(result);
    }
}

File: attached_assets/Starter_code/Java_starter_code/src/main/java/com/example/insurance/controller/UnderwritingController.java
package com.example.insurance.controller;

import com.example.insurance.service.UnderwritingService;
import org.springframework.web.bind.annotation.*;

record EvaluateRequest(int age, String product, int priorClaims) {}

@RestController
@RequestMapping("/api/underwriting")
public class UnderwritingController {
    private final UnderwritingService service;
    public UnderwritingController(UnderwritingService service) { this.service = service; }

    @PostMapping("/evaluate")
    public UnderwritingService.Result evaluate(@RequestBody EvaluateRequest req) {
        return service.evaluate(req.age(), req.product(), req.priorClaims());
    }
}


